plugins/rikki/heroeslounge/classes/helpers/TimezoneHelper.php

```diff
@@ -0,0 +1,52 @@
+<?php namespace Rikki\Heroeslounge\Classes\Helpers;
+
+use Auth;
+use Config;
+use Session;
+
+class TimezoneHelper
+{
+    private const TIMEZONE_KEY = 'timezone';
+
+    public static $defaultTimezone;
+
+    private static function defaultTimezone()
+    {
+        if (!self::$defaultTimezone)
+        {
+            self::$defaultTimezone = Config::get('app.timezone', 'Europe/Berlin');
+        }
+        return self::$defaultTimezone;
+    }
+
+    public static function getTimezone()
+    {
+        return Session::get(self::TIMEZONE_KEY, self::defaultTimezone());
+    }
+
+    public static function setTimezone()
+    {
+        if (isset($_POST[self::TIMEZONE_KEY])) {
+            $timezoneName = $_POST[self::TIMEZONE_KEY];
+        } else {
+            $timezoneName = self::defaultTimezone();
+        }
+        if (!in_array($timezoneName, timezone_identifiers_list())) {
+            $timezoneName = self::defaultTimezone();
+        }
+        Session::put(self::TIMEZONE_KEY, $timezoneName);
+
+        $user = Auth::getUser();
```

Comment:
turns out, this never gets called because by the time you are logged in, the timezone is already set

Reply:
In my experience, one can stay logged in on the page for an indefinite amount of time

Reply:
Yeah the login is permanent, but the session is per browser session, I think.